From edef11f84ba0f7f8025904bcd2ceaf88a4aa952d Mon Sep 17 00:00:00 2001
From: Vasilij Schneidermann <mail@vasilij.de>
Date: Fri, 8 Jun 2018 12:27:30 +0200
Subject: [PATCH] Use python2

---
 Makefile | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/Makefile b/Makefile
index 5494f26..598ab43 100644
--- a/Makefile
+++ b/Makefile
@@ -81,7 +81,7 @@ $(OBJECTS) : $(HEADERS)
 aweio.o: aweio.c scanner.inc
 
 scanner.inc: scanner.py
-	python scanner.py
+	python2 scanner.py
 
 libawe.a: $(OBJECTS)
 	rm -f libawe.a
@@ -148,26 +148,26 @@ manpages: awe.1 awe.mk.7
 webpages: manpages awe.html awe.1.html awe.mk.7.html INSTALL.html
 
 awe.1 : man.py awe.1.src VERSION
-	python man.py awe.1.src awe.1 \
+	python2 man.py awe.1.src awe.1 \
                       VERSION="$(shell cat VERSION)" \
                       BINDIR="$(BINDIR)" LIBDIR="$(LIBDIR)" DOCDIR="$(DOCDIR)" INCDIR="$(INCDIR)"
 
 awe.mk.7 : man.py awe.mk.7.src VERSION
-	python man.py awe.mk.7.src awe.mk.7 VERSION="$(shell cat VERSION)" INCDIR="$(INCDIR)"
+	python2 man.py awe.mk.7.src awe.mk.7 VERSION="$(shell cat VERSION)" INCDIR="$(INCDIR)"
 
 #
 
 awe.1.html: awe.1 htmltext.py
-	MANWIDTH=80 man ./awe.1 | python htmltext.py "awe(1): Awe ALGOL W compiler man page" > awe.1.html
+	MANWIDTH=80 man ./awe.1 | python2 htmltext.py "awe(1): Awe ALGOL W compiler man page" > awe.1.html
 
 awe.mk.7.html: awe.mk.7 htmltext.py
-	MANWIDTH=80 man ./awe.mk.7 | python htmltext.py "awe.mk(7): Awe ALGOL W Makefile man page" > awe.mk.7.html
+	MANWIDTH=80 man ./awe.mk.7 | python2 htmltext.py "awe.mk(7): Awe ALGOL W Makefile man page" > awe.mk.7.html
 
 awe.html: awe.txt htmltext.py
-	python htmltext.py "awe.txt: Awe ALGOL W compiler documentation file" < awe.txt > awe.html
+	python2 htmltext.py "awe.txt: Awe ALGOL W compiler documentation file" < awe.txt > awe.html
 
 INSTALL.html: INSTALL htmltext.py
-	python htmltext.py "INSTALL: Awe ALGOL W compiler installation instructions" < INSTALL > INSTALL.html
+	python2 htmltext.py "INSTALL: Awe ALGOL W compiler installation instructions" < INSTALL > INSTALL.html
 
 
 # ------------------------------------------------------------------------------
-- 
2.17.0

